library/src/cache/patch_manager.rs

```diff
@@ -229,6 +234,20 @@ impl PatchManager {
             }
         }
 
+        if let Some(public_key) = &self.patch_public_key {
+            // If we have a public key, verify that the patch has a signature
+            let patch_signature = patch
+                .signature
+                .clone()
+                .context("Patch signature is missing")?;
+
+            // Check that the signature is valid.
+            let patch_hash = signing::hash_file(&artifact_path)?;
+            signing::check_signature(&patch_hash, &patch_signature, public_key)?;
+        } else {
+            info!("No public key provided, skipping signature verification");
+        }
```

Comment:
Not sure where the hash check is.

I would expect the process to be:
1. If we have a public key, check the signature on the hash matches.
2. Hash the patch
3. Check the hash matches.

Presumably that's already being done, just spread across other methods?

Reply:
> Not sure where the hash check is.

We don't currently check the hash for equality, and I'm not sure we need to—any change to the hash would cause the signature to become invalid.

The process is:

1. If we have a public key, hash the patch file
2. Check that the signature matches the hash

Reply:
I think I/we need to write out exactly what we're singing in plain language.  Maybe I'll add that to my in-progress blog post tomorrow.